### PR TITLE
MAINT: Start applying rruff/flake8-pie rules (PIE)

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
In preparation for https://github.com/numpy/numpy/issues/24994.

### CODE REVIEW

**Review by mhvk — APPROVED** [review]
Also a definite improvement (and not much work to review), thanks.

**Comment by mhvk** [comment]
Though you'll need to fix the indentation error!

File: doc/postprocess.py [hunk 1]
```diff
@@ -34,12 +34,15 @@ def process_tex(lines):
...
-            ):
+        if line.startswith(
+            (
+                "\\section{numpy.",
+                "\\subsection{numpy.",
+                "\\subsubsection{numpy.",
+                "\\paragraph{numpy.",
+                "\\subparagraph{numpy.",
+            )
+        ):
```
Comment (charris) [inline doc/postprocess.py:45]: I'm not completely happy with this style, but OK.

Comment (DimitriPapadopoulos) [reply doc/postprocess.py:45]: I tried to mimic the initial style, but the linter complained or the result was not readable. This format is black / ruff style, pretty standard nowadays. Initially I  didn't like that style either, but it is readable and can be consistently applied in all contexts. I am happy to change to try again to adapt the initial style.

File: doc/postprocess.py [hunk 2]
```diff
@@ -34,13 +34,13 @@ def process_tex(lines):
     """
     new_lines = []
     for line in lines:
-        if (line.startswith(r'\section{numpy.')
-            or line.startswith(r'\subsection{numpy.')
-            or line.startswith(r'\subsubsection{numpy.')
-            or line.startswith(r'\paragraph{numpy.')
-            or line.startswith(r'\subparagraph{numpy.')
-            ):
-            pass # skip!
```
Comment (DimitriPapadopoulos) [inline doc/postprocess.py:43]: The `skip` comment after `pass` did not seem helpful.

File: doc/postprocess.py (hunk 1 above)
Comment (charris) [reply doc/postprocess.py]: Part of my problem was that I couldn't think of anything that was a big improvement. Your new fix at least clearly shows the condition of the `if`, which always helps when reading the code.

**Comment by charris** [comment]
Thanks @DimitriPapadopoulos .

**Pull request merged**